tools/dm/data-synchronization-features.md

```diff
@@ -0,0 +1,452 @@
+- `rule-2` 用来同步匹配上 `schema-pattern: "test_*"` 的库的 DDLs，例如 `create/drop schema xx`。
+
+> **注意:**
+>
+> - 如果下游 TiDB `schema: test` 已经存在， 并且不会被删除则可以省略 `rule-2`。
+> - 如果下游 TiDB `schema: test` 不存在，只设置了 `rule_1` 则同步会报错 `schema test doesn't exist`。
+
+```yaml
+  rule-1:
+    schema-pattern: "test_*"
+    table-pattern: "t_*"
+    target-schema: "test"
+    target-table: "t"
+  rule-2:
+    schema-pattern: "test_*"
+    target-schema: "test"
+```
+
+#### 分库合并
+
+假设存在分库场景，将上游两个 MySQL 实例 `test_{1,2,3...}`.`t_{1,2,3...}` 同步到下游 TiDB 的 `test`.`t_{1,2,3...}`，创建一条路由规则即可：
+
+```yaml
+  rule-1:
+    schema-pattern: "test_*"
+    target-schema: "test"
+```
+
+#### 错误的 table routing
+
+假设存在下面两个路由规则，`test_1_bak`.`t_1_bak` 可以匹配上 `rule-1` 和 `rule-2`， 违反 table 路由的限制而报错。
+
+```yaml
+  rule-1:
+    schema-pattern: "test_*"
+    table-pattern: "t_*"
+    target-schema: "test"
+    target-table: "t"
+  rule-2:
+    schema-pattern: "test_1_bak"
+    table-pattern: "t_1_bak"
+    target-schema: "test"
+    target-table: "t_bak"
+```
+
+## Black and white table lists
+
+上游数据库实例表的黑白名单过滤规则，可以用来过滤或者只同步某些 `database/table` 的所有操作。
+
+### 配置参数
+
+```yaml
+black-white-list:
+  rule-1:
+    do-dbs: ["~^test.*"]         # 以 ~ 字符开头，表示规则是正则表达式
+​    ignore-dbs: ["mysql"]
+    do-tables:
+    - db-name: "~^test.*"
+      tbl-name: "~^t.*"
+    - db-name: "test"
+      tbl-name: "t"
+    ignore-tables:
+    - db-name: "test"
+      tbl-name: "log"
+```
+
+### 参数解释
+
+- `do-dbs` 要同步的库的白名单
+- `ignore-dbs` 要同步的库的黑名单
+- `do-tables` 要同步的表的白名单
+- `ignore-tables` 要同步的表的黑名单
+- 上面黑白名单中 以 ~ 字符开头名称为[正则表达式](https://golang.org/pkg/regexp/syntax/#hdr-Syntax)。
+
+### 过滤规则
+
+判断 table `test`.`t` 是否应该被过滤的过滤流程如下：
+
+1. 首先 *schema 过滤判断*
+
+    - 如果 `do-dbs` 不为空，判断 `do-dbs` 中是否存在一个匹配的 schema。
+
+        - 如果存在，则进入 *table 过滤判断*。
+        - 如果不存在，则过滤 `test`.`t`。
+
+    - 如果 `do-dbs` 为空并且 `ignore-dbs` 不为空，判断 `ignore-dbs` 中是否存在一个匹配的 schema。
+
+        - 如果存在，则过滤 `test`.`t`。
+        - 如果不存在，则进入 *table 过滤判断*。
+
+    - 如果 `do-dbs` 和 `ignore-dbs` 都为空，则进入 *table 过滤判断*。
+
+2. 进行 *table 过滤判断*:
+
+    1. 如果 `do-tables` 不为空，判断 `do-tables` 中是否存在一个匹配的 table。
+
+        - 如果存在，则同步 `test`.`t`。
+        - 如果不存在，则过滤 `test`.`t`。
+
+    2. 如果 `ignore-tables` 不为空，判断 `ignore-tables` 中是否存在一个匹配的 table。
+
+        - 如果存在，则过滤 `test`.`t`.
+        - 如果不存在，则同步 `test`.`t`。
+
+    3. 如果 `do-tables` 和 `ignore-tables` 都为空，则同步 `test`.`t`。
+
+> **注意:**  判断 schema `test` 是否被过滤，只进行 *schema 过滤判断*
+
+### 使用示例
+
+假设上游 MySQL 实例包含下面的 tables:
+
+```
+`logs`.`messages_2016`
+`logs`.`messages_2017`
+`logs`.`messages_2018`
+`forum`.`users`
+`forum`.`messages`
+`forum_backup_2016`.`messages`
+`forum_backup_2017`.`messages`
+`forum_backup_2018`.`messages`
+```
+
+配置如下:
+
+```yaml
+black-white-list:
+  bw-rule:
+    do-dbs: ["forum_backup_2018", "forum"]
+    ignore-dbs: ["~^forum_backup_"]
+    do-tables:
+    - db-name: "logs"
+      tbl-name: "~_2018$"
```

Comment:
this `do-tables` rule is useless because schema `log` has been filtered. It may confuse users, shall we remove this rule?

Reply:
keep it